docs/cloud/connecting-services.mdx

```diff
@@ -5,45 +5,298 @@ icon: "link"
+The API key starts with `api_`. Put it in a file named `token` and add the secret:
 ```bash
-restate cloud login
+kubectl create secret generic my-cloud-environment-secret \
+  --from-file=token=./token -n restate-operator
 ```
 
-Set up a new CLI profile for your environment:
-```bash
-restate cloud env configure
+### Set up a secure tunnel
+Set up a secure tunnel between the Kubernetes cluster and Restate Cloud with the `RestateCloudEnvironment` CRD:
+
+```yaml
+apiVersion: restate.dev/v1beta1
+kind: RestateCloudEnvironment
+metadata:
+  name: my-cloud-environment
+spec:
+    environmentId: env_201j05r9g0f12ygtphdszbb4scp
+    signingPublicKey: publickeyv1_BBuEJnx28hpGb5Ky6qpvuXQG4aVoWBnubJtHXpznzgQk
+    region: eu
+    authentication:
+        secret:
+        name: my-cloud-environment-secret
+        key: token
 ```
 
-Tell the CLI to use the new environment:
-```bash
-restate config use-env <name>
+- The `environmentId` can be found in the Restate Cloud UI in the top left corner when you select your environment.
+This ID starts with `env_`.
+
+- The `signingPublicKey` can be found in the Restate Cloud UI under `Developers`. Scroll down to the `Security` section to find the public key under `HTTP endpoints`.
+The public key starts with `publickeyv1_`.
+
+- Set the `region` field to the region your Restate Cloud environment is hosted in. Possible values are `us` and `eu`.
+You can find the region next to your environment ID in the Restate Cloud UI.
+
+View the full spec of the `RestateCloudEnvironment` CRD as [pkl file](https://github.com/restatedev/restate-operator/blob/main/crd/RestateCloudEnvironment.pkl) or less-readable [yaml file](https://github.com/restatedev/restate-operator/blob/main/crd/restatecloudenvironments.yaml).
+
+<Info>
+    [Learn more about how the tunnel works below.](/cloud/connecting-services#how-does-the-tunnel-work)
+</Info>
+
+### Create your RestateDeployment
+
+Create a `RestateDeployment` manifest for your service, as described in the [deployment documentation](/services/deploy/kubernetes#deployment-with-restate-operator), and specify the cloud environment to register with in the `spec`:
+```yaml
+spec:
+  restate:
+    register:
+      cloud: my-cloud-environment
+```
+
+Once you apply the manifest, the Restate Operator will set up the secure tunnel and automatically register your service with Restate Cloud.
+
+
+## Serverless Functions and other public endpoints
+
+Restate can invoke your services over the public internet.
+This is useful if you are deploying your services as serverless functions, e.g. on Vercel, Cloudflare Workers, or Deno Deploy.
+Or if you have services behind public endpoints that you want to connect to Restate Cloud.
+
+<Info>
+    For AWS Lambda, please see the [AWS Lambda functions](#aws-lambda-functions) section below.
+</Info>
+
+It is important to secure access to your publicly accessible services so that only Restate can call them.
```

Comment:
Maybe this sounds less scary?
```suggestion
It is important to secure access to your publicly routable services so that only the designated Restate environment can call them.
```